### PR TITLE
Helm chart for nats-operator

### PR DESCRIPTION
**Pull request by richerlariviere**
Following our discussion (see issue #61 ), I decided to create a helm chart for nats-operator. I had some experience with Helm and my team is using NATS-operator (thanks a lot by the way). Despite my little understanding of NATS I think that this chart can cover the basic usage of NATS. So here is what I did so far:

- [x] Test cluster with authentication enabled
- [x] Test cluster with RBAC enabled
- [x] Test with 1 or more users
- [x] Test cluster with TLS 
- [x] Test cluster with configuration reload

 I'd like to have your feedback for other tests I could do.

I suggest that we don't upload this Helm on [KubeApps](https://hub.kubeapps.com/) until it's mature and well-tested.

**Thank you for your feedback!** 😀

### CODE REVIEW

File: helm/nats-operator/README.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+NATS Operator manages [NATS](https://nats.io/) clusters atop [Kubernetes](http://kubernetes.io), automating their creation and administration. With NATS-operator you can benefits from the flexibility brought by the Kubernetes operator pattern. It means less juggling between manifests and a few handy features like automatic configuration reload.
+
+If you want to manage NATS entirely by yourself and have more control over your NATS cluster, you can always use the original [NATS](https://github.com/helm/charts/tree/master/stable/nats) Helm chart.
+
+
+## Prerequisites
+
+Basic
+- Kubernetes 1.4+ with Beta APIs enabled
+- PV provisioner support in the underlying infrastructure
```
Comment (wallyqs) [inline helm/nats-operator/README.md]: sorry not following which Kubernetes feature is this one... we should just make the prerequisites same as in the nats operator readme (`Kubernetes v1.8+`, etc...)

File: helm/nats-operator/README.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+  #username: 
+  #password: 
+
+  # values.yaml
+  users:
+    - username: "my-user"
+      password: "T0pS3cr3t"
+    - username: "my-user-2"
+      password: "MyS3cr3tP4ssw0rd"
```
Comment (wallyqs) [inline helm/nats-operator/README.md]: we could also declare `permissions:` here but could be done in a different PR

File: helm/nats-operator/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+              fieldPath: metadata.namespace
+        - name: MY_POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+          {{- if .Values.livenessProbe.enabled }}
+          livenessProbe:
+            httpGet:
+              path: /
+              port: monitoring
```
Comment (wallyqs) [inline helm/nats-operator/templates/deployment.yaml]: Not sure that the NATS Operator has a monitoring port for liveness/readiness probe (may have missed it... if so we should include in the examples...)

File: helm/nats-operator/README.md (hunk 1 above)
Comment (richerlariviere) [reply helm/nats-operator/README.md]: I changed it si we only keep `Kubernetes v1.8+`. I don't know what I was thinking. I probably assumed wrongly that NATS and NATS-operator shared the same requirements in terms of Kubernetes configuration. I just pushed a [commit](c3a6cb5417f211920a671d4b718dd3c5ea3d9398) to fix that.

File: helm/nats-operator/README.md (hunk 2 above)
Comment (richerlariviere) [reply helm/nats-operator/README.md]: Yeah of course! I'll remind myself to create an issue after this one is merged.

File: helm/nats-operator/templates/deployment.yaml (hunk 3 above)
Comment (wallyqs) [reply helm/nats-operator/templates/deployment.yaml]: Ok so we can use something for the readiness probe, adding to the examples: https://github.com/nats-io/nats-operator/pull/63

Comment (richerlariviere) [reply helm/nats-operator/templates/deployment.yaml]: I pushed a new commit changing the port to `8080`. I tested it in my side and I can poke `:8080/readyz` succesfully

File: helm/nats-operator/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: MY_POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+          ports:
+          - name: readyz 
+            port: 8080
```
Comment (wallyqs) [inline helm/nats-operator/templates/deployment.yaml]: I think this one has to be `containerPort`? https://github.com/nats-io/nats-operator/blob/68079dc2093aba61711a8a19ae0989bd181b9cf4/example/deployment-rbac.yaml#L29-L35

```
        ports:
        - name: readyz
          containerPort: 8080
```

File: helm/nats-operator/README.md (hunk 2 above)
Comment (richerlariviere) [reply helm/nats-operator/README.md]: Issue created https://github.com/nats-io/nats-operator/issues/67

**Comment by richerlariviere** [comment]
Added support for permissions. Fix #67

**Pull request merged**